server/routes/createGoal/routerRequestHandlers.ts

```diff
@@ -0,0 +1,30 @@
+import { NextFunction, Request, Response } from 'express'
+import logger from '../../../logger'
+
+/**
+ * A module exporting request handler functions to support ensuring the Create A Goal process has been followed
+ * in the correct sequence - IE. pages are not being called out of sequence.
+ */
+
+/**
+ * Request handler function to check the CreateGoalForm exists in the session for the prisoner reference in the
+ * request URL.
+ */
+const checkCreateGoalFormExistsInSession = async (req: Request, res: Response, next: NextFunction) => {
+  if (!req.session.createGoalForm) {
+    logger.warn(
+      `No CreateGoalForm object in session - user attempting to navigate to path ${req.path} out of sequence. Redirecting to start of Create Goal journey.`,
+    )
+    res.redirect(`/plan/${req.params.prisonNumber}/goals/create`)
+  } else if (req.session.createGoalForm.prisonNumber !== req.params.prisonNumber) {
+    logger.warn(
+      `CreateGoalForm object in session references a different prisoner. Redirecting to start of Create Goal journey.`,
```

Comment:
Minor - this is ok, but perhaps it could be expressed in less technical "business" terms (i.e. so that someone who doesn't know what the `CreateGoalForm` is might be able to understand the issue just from the logs)?

Having said that, I'm finding it difficult to choose the right words! 😆

Reply:
Thats a fair shout.
I think we'll be adding a few more of these type of handler functions (fairly soon we will do the one that I hinted at in this PR where we need to check you have at least 1 AddStepForm when you arrive on Add Note)
So we can consider how we word these messages etc then I guess